pkg/git/gogit/checkout.go

```diff
@@ -33,29 +33,30 @@ import (
 	"github.com/fluxcd/source-controller/pkg/git"
 )
 
-func CheckoutStrategyForRef(ref *sourcev1.GitRepositoryRef) git.CheckoutStrategy {
+func CheckoutStrategyForRef(ref *sourcev1.GitRepositoryRef, recurseSubmodules bool) git.CheckoutStrategy {
```

Comment:
I'd consider using the "option" pattern for this and making it varargs, so existing uses won't break. Not a deal-breaker, more a courtesy :-)

Reply:
Can you provide an example?

Reply:
Here's an example from controller-runtime: https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/client#MergeFromWithOptions.

The general pattern is to have a varargs argument taking arguments which will operate on the value being constructed. The effect is like having keyword arguments in Python; you don't have to supply them. Specifically, here, it means existing call sites don't have to be altered to provide the new argument. Is this a huge deal? Not really, it just makes an optional, boolean argument a bit more palatable. (The effect is greater when there's many such arguments, of course).

Reply:
I've added a struct to hold all the checkout options, the varargs feels too complicated for me.

Reply:
OK. This way doesn't avoid changing call sites, but it is neater since it works like keyword arguments (albeit with a keyword argument that is mandatory).